### PR TITLE
servicebus jms autoconfiguration enabled fix

### PR DESCRIPTION
**Pull request by jialigit**
# Description

Solves [#29232](https://github.com/Azure/azure-sdk-for-java/issues/29232)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,14 @@ This section includes changes in `spring-cloud-azure-autoconfigure` module.
 #### Dependency Updates
 - Upgrade spring-security to 5.6.4 to address [CVE-2022-22978](https://spring.io/blog/2022/05/15/cve-2022-22978-authorization-bypass-in-regexrequestmatcher) [#29304](https://github.com/Azure/azure-sdk-for-java/pull/29304).
 
+### spring-cloud-azure-starter-servicebus-jms
```
Comment (saragluna) [inline sdk/spring/CHANGELOG.md:24]: The jms is part of the Spring Cloud Azure Autoconfigure

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java [hunk 2]
```diff
@@ -42,11 +44,18 @@
...
-@EnableConfigurationProperties({ AzureServiceBusJmsProperties.class, JmsProperties.class })
+@EnableConfigurationProperties(JmsProperties.class)
 @Import({ ServiceBusJmsConnectionFactoryConfiguration.class, ServiceBusJmsContainerConfiguration.class })
 public class ServiceBusJmsAutoConfiguration {
 
+    @ConfigurationProperties(prefix = AzureServiceBusJmsProperties.PREFIX)
+    @Bean
+    AzureServiceBusJmsProperties azureServiceBusJmsProperties() {
+        return new AzureServiceBusJmsProperties();
+    }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java:57]: What's the difference of writing it here and putting it in the `@EnableConfigurationProperties`?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java [hunk 3]
```diff
@@ -16,14 +16,18 @@
...
-@ConfigurationProperties(prefix = AzureServiceBusJmsProperties.PREFIX)
 public class AzureServiceBusJmsProperties implements InitializingBean {
 
     /**
      * Service Bus JMS properties prefix.
      */
     public static final String PREFIX = "spring.jms.servicebus";
 
+    /**
+     * Whether to enable Servive Bus autoconfiguration.
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java:27]: Service Bus JMS

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java (hunk 3 above)
Comment (jialigit) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/properties/AzureServiceBusJmsProperties.java]: updated

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java (hunk 2 above)
Comment (jialigit) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jms/ServiceBusJmsAutoConfiguration.java]: reverted.

File: sdk/spring/CHANGELOG.md (hunk 1 above)
Comment (jialigit) [reply sdk/spring/CHANGELOG.md]: updated

**Comment by jialigit** [comment]
@saragluna we have a change in the autoconfiguration which bring error in revapi check. This may lead the ci fail.

**Comment by saragluna** [comment]
@jialigit we're okay to suppress such changes in revapi

**Comment by jialigit** [comment]
@saragluna , we ignore the difference in the revapi.json:
adding the following rule for service bus jms autoconfiguration.

```json  
   {
          "code": "java.annotation.added",
          "new": "class com.azure.spring.cloud.autoconfigure.jms.ServiceBusJmsAutoConfiguration",
          "justification": "Fixes a bug."
        }
```

**Review by saragluna — APPROVED** [review]

**Pull request merged**